pandas/io/excel/_openpyxl.py

```diff
@@ -451,3 +462,280 @@ def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0,
+        if header is not None and is_list_like(header):
+            header_names = []
+            control_row = [True] * len(data[0])
+
+            for row in header:
+                if is_integer(skiprows):
+                    row += skiprows
+
+                data[row], control_row = _fill_mi_header(data[row],
+                                                         control_row)
+
+                if index_col is not None:
+                    header_name, _ = _pop_header_name(data[row], index_col)
+                    header_names.append(header_name)
+        return header_names
+
+    @staticmethod
+    def _handle_convert_float(series, convert_float):
+        """Handle the convert_float keyword."""
+        # attempt to convert object columns to integer. Only because this
+        # is implicitly done when reading and excel file with xlrd, that
+        # behaviour is replicated here.
+
+        if series.dtype == object:
+            try:
+                series = series.astype('int64')
+            except (ValueError, TypeError):
+                try:
+                    series = series.astype('float64')
+                except (ValueError, TypeError):
+                    return series
+        elif (convert_float
+                and series.dtype >= float
+                and all(series % 1 == 0)):
+            series = series.astype('int64')
+        elif not convert_float:
+            if series.dtype >= int:
+                series = series.astype('float64')
+        return series
+
+    @staticmethod
+    def _handle_index_col(frame, index_col):
+        column_names = frame.columns.values
+        if index_col is None:
+            return frame
+        if is_list_like(index_col):
+            if any(isinstance(i, str) for i in index_col):
+                # TODO: see if there is already a method for this in
+                # pandas.io.parsers
+                frame = frame.set_index(index_col)
+                if len(index_col) == 1:
+                    # TODO: understand why this is needed
+                    raise TypeError("list indices must be integers.*, not str")
+            else:
+                frame = frame.set_index([column_names[i] for i in index_col])
+        else:
+            if isinstance(index_col, str):
+                frame = frame.set_index(index_col)
+            else:
+                frame = frame.set_index(column_names[index_col])
+        return frame
+
+    def get_sheet_by_name(self, name):
+        return self.book[name]
+
+    def get_sheet_by_index(self, index):
+        return self.book.worksheets[index]
+
+    def _replace_type_error_with_nan(self, rows):
+        try:
+            from openpyxl.cell.cell import TYPE_ERROR
+        except ImportError:
+            # Work around for import error in Linux py35_compat test that I
+            # can't reproduce
+            TYPE_ERROR = 'e'
+
+        nan = float('nan')
+        for row in rows:
+            yield [nan
```

Comment:
Why yield here? It looks like the only place this gets called immediately turns it into a list anyway

Reply:
Just a personal preference for generators, but indeed this doesn't add anything here

Reply:
Yea if it doesn't add anything go ahead and remove